<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ane Liv Berthelsen">
<meta name="dcterms.date" content="2023-11-20">

<title>Sustainable genetics project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="SGP_files/libs/clipboard/clipboard.min.js"></script>
<script src="SGP_files/libs/quarto-html/quarto.js"></script>
<script src="SGP_files/libs/quarto-html/popper.min.js"></script>
<script src="SGP_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="SGP_files/libs/quarto-html/anchor.min.js"></script>
<link href="SGP_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="SGP_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="SGP_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="SGP_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="SGP_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#abstract" id="toc-abstract" class="nav-link active" data-scroll-target="#abstract">Abstract</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#materials-and-methods" id="toc-materials-and-methods" class="nav-link" data-scroll-target="#materials-and-methods">Materials and Methods</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a></li>
  <li><a href="#supporting-information" id="toc-supporting-information" class="nav-link" data-scroll-target="#supporting-information">Supporting Information</a>
  <ul class="collapse">
  <li><a href="#description" id="toc-description" class="nav-link" data-scroll-target="#description">Description</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data">Data</a></li>
  <li><a href="#pcr-program" id="toc-pcr-program" class="nav-link" data-scroll-target="#pcr-program">PCR Program</a></li>
  <li><a href="#missing-data" id="toc-missing-data" class="nav-link" data-scroll-target="#missing-data">Missing data</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Sustainable genetics project</h1>
<p class="subtitle lead">Script</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Ane Liv Berthelsen </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 20, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="abstract" class="level1">
<h1>Abstract</h1>
<p>Scientific research is a massive endeavor, consuming vast quantities of single-use plastics. With an ongoing global plastic crisis at our hands, we cannot hide behind the importance of scientific progress. A multitude of possible solutions from shifting to reusable materials to generating shared campus-wide laboratory inventories are being proposed. Rethinking existing laboratory protocols is often encouraged, but it is rarely based on empirical testing. We designed an experimental set-up to evaluate the re-useability of 96 microwell plates in the context of an existing microsatellite genotyping protocol. In our standard procedure 11 microplates are required to genotype 96 samples. The samples are distributed from one plate to five plates for the first step (PCR amplification) and transferred to five new plates in the second step (fragment analysis). Due to the risk of contamination, the plates are usually single-use. We cleaned the plates using an environmentally friendly approach and systematically reused the plates in either the first step (Re-used PCR plate) or the second step (Re-used detection plate) of the protocol. We included an internal control treatment to establish baseline error rates. We found no detectable difference in quality of either scoring or final genotyping for the ‘Re-used detection plate’ treatment. Whereas the ‘Re-used PCR plates’ led to a decreased scoring and a 10-fold increase in genotype errors. We conclude, that washing and re-using 96 microwell plates for fragment analysis can decrease the use of single-use plastics in laboratories, thereby saving costs and improving sustainability in the lab.</p>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>We set out to explore the re-useability of 96-micro well plate with the aim to test the following hypothesis: 1) 96 microwell plates can be cleaned and reused within the context of microsatellite genotyping, 2) the genotype error rate will be higher for plates reused for PCR amplification.</p>
</section>
<section id="materials-and-methods" class="level1">
<h1>Materials and Methods</h1>
<p>We designed an experimental setup to assess the re-usability of 96-micro well plates within the context of microsatellite genotyping. The setup contained 4 treatments: standard procedure, internal control, re-used PCR plate and re-used detection plate (See <a href="#fig-Schematic">Figure&nbsp;1</a>). In the standard procedure total genomic DNA is distributed from a rack to be PCR amplified in five separate multiplexed reactions on ‘PCR plates’ (PCR trays ROTILABO® 96 well, Standard, half frame, Roth Selection, Karlsruhe, Germany). Thereafter, the samples are resolved on ‘detection plates’ (Fisherbrand™ 96-Well Semi-Skirted PCR Plates, Thermo Fisher Scientific, Waltham, MA, USA). After the initial use, the plates were washed using the following steps; first, each plate was individually rinsed with distilled water and emptied 10 times, before submerging in soap water for two hours. After soaking, each plate was again rinsed and emptied 10 times before being left on a paper towel over night to dry. The re-used PCR plates and detection plates all originated from the ‘standard procedure’ round. We allocated the re-used plates to another rack than the original.</p>
<div id="fig-Schematic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/SGPsetup5.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Schematic overview</figcaption>
</figure>
</div>
<section id="microsatellite-genotyping" class="level4">
<h4 class="anchored" data-anchor-id="microsatellite-genotyping">Microsatellite genotyping</h4>
<p>285 tissue samples were collected from Bird Island, South Georgia (54°00024.800 S, 38°03004.100 W) during the austral summer of 2020-2021 (274 samples), 2015-2016 (9 positive controls), and 2 samples on rack 154 from 2006-2007. Tissue samples were collected from the flipper or umbilical cord of Antarctic fur seal pups and stored in 20% dimethyl sulphoxide saturated with sodium chloride at -20°C. The DNA was extracted using a standard chloroform-isoamylalcohol protocol and genotyped at 39 microsatellite loci separated into five multiplexed reactions for improved scoring. PCR amplification was performed using a Type It Kit (Qiagen) with the following program: 5 min denaturation at 94°C; 28 cycles of 30 sec at 94°C, 90 sec at mix-specific temperature (Supplementary table 1), and 30 sec at 72°C, finished by an extension of 30 min at mix-specific temperature. The PCR product was transferred to and diluted on detection plates before being resolved by electrophoresis on an ABI 3730xl capillary sequencer (Applied Biosystems, Thermo Fisher Scientific, Waltham, MA, USA). Each plate contained three positive controls to ensure standardisation of microsatellite scoring across plates. Allele sizes were scored automatically using GeneMarker v. 2.6.2 (SoftGenetics, LLC., State College, PA, USA) and manually inspected and corrected when necessary. Following customary procedure, any locus with more than 30% missing data was attempted to be gapfilled separately to minimize missing data.</p>
</section>
<section id="data-evaluation" class="level4">
<h4 class="anchored" data-anchor-id="data-evaluation">Data evaluation</h4>
<p>We calculated the allelic and genotype error rate between the different treatments using the ‘standard procedure’ treatment as reference. The ‘internal control’ treatment was included in the experimental setup to compute a baseline for error rates. The comparison between the different treatments and the standard procedure yielded a list of mismatches. In the cases where the mismatch stemmed from the ‘re-used PCR plate’ or ‘re-used detection plate’ treatment, the new genotype was compared to the original genotype. Here, original genotype refers to the genotype resolved from the same well in the first use of the plate. If the genotype matched the original genotype, the mismatch was marked as potential case of contamination. Additionally, we calculated the rate of missing data for all treatments.</p>
<p>For each sample we noted whether the genotype matched with the ‘standard procedure’ per loci and for all treatments whether the genotype was scored or not. This yielded two datasets referred to as ‘match’ and ‘missing genotype’ data respectively. We ran separate Bayesian binary logistic regression mixed models with treatment included as several indicator variables for match data and missing genotypes data. All statistical analyses were implemented in R version 4.2.1 with Rstudio version 2023.09.1+494.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-allelemismatch" class="anchored">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<caption>Table&nbsp;1: Error rates</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Rack 1</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rack 2</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Treatment</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">No. of mismatches</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Allelic error rate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Genotype error rate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">R154_R1</td>
<td style="text-align: left;">R154_R2</td>
<td style="text-align: left;">Internal control</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">0.0012980</td>
<td style="text-align: right;">0.0020190</td>
</tr>
<tr class="even">
<td style="text-align: left;">R155_R1</td>
<td style="text-align: left;">R155_R2</td>
<td style="text-align: left;">Internal control</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">0.0028039</td>
<td style="text-align: right;">0.0047666</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R156_R1</td>
<td style="text-align: left;">R156_R2</td>
<td style="text-align: left;">Internal control</td>
<td style="text-align: right;">45</td>
<td style="text-align: right;">0.0064157</td>
<td style="text-align: right;">0.0128315</td>
</tr>
<tr class="even">
<td style="text-align: left;">R154_R1</td>
<td style="text-align: left;">R154_R3</td>
<td style="text-align: left;">Re-used detection plate</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.0021564</td>
<td style="text-align: right;">0.0031627</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R155_R1</td>
<td style="text-align: left;">R155_R3</td>
<td style="text-align: left;">Re-used detection plate</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">0.0015127</td>
<td style="text-align: right;">0.0027503</td>
</tr>
<tr class="even">
<td style="text-align: left;">R156_R1</td>
<td style="text-align: left;">R156_R3</td>
<td style="text-align: left;">Re-used detection plate</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">0.0037303</td>
<td style="text-align: right;">0.0068867</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R154_R1</td>
<td style="text-align: left;">R154_R4</td>
<td style="text-align: left;">Re-used PCR plate</td>
<td style="text-align: right;">139</td>
<td style="text-align: right;">0.0239821</td>
<td style="text-align: right;">0.0472740</td>
</tr>
<tr class="even">
<td style="text-align: left;">R155_R1</td>
<td style="text-align: left;">R155_R4</td>
<td style="text-align: left;">Re-used PCR plate</td>
<td style="text-align: right;">113</td>
<td style="text-align: right;">0.0173260</td>
<td style="text-align: right;">0.0337320</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R156_R1</td>
<td style="text-align: left;">R156_R4</td>
<td style="text-align: left;">Re-used PCR plate</td>
<td style="text-align: right;">68</td>
<td style="text-align: right;">0.0105231</td>
<td style="text-align: right;">0.0194986</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-contamination" class="anchored">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<caption>Table&nbsp;2: Potential contamination cases</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Rack</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Control</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Cases</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Treatment</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">R154_R3</td>
<td style="text-align: left;">R156_R1</td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">Re-used detection plate</td>
</tr>
<tr class="even">
<td style="text-align: left;">R155_R3</td>
<td style="text-align: left;">R154_R1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Re-used detection plate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R156_R3</td>
<td style="text-align: left;">R155_R1</td>
<td style="text-align: right;">8</td>
<td style="text-align: left;">Re-used detection plate</td>
</tr>
<tr class="even">
<td style="text-align: left;">R154_R4</td>
<td style="text-align: left;">R156_R1</td>
<td style="text-align: right;">105</td>
<td style="text-align: left;">Re-used PCR plate</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R155_R4</td>
<td style="text-align: left;">R154_R1</td>
<td style="text-align: right;">95</td>
<td style="text-align: left;">Re-used PCR plate</td>
</tr>
<tr class="even">
<td style="text-align: left;">R156_R4</td>
<td style="text-align: left;">R155_R1</td>
<td style="text-align: right;">53</td>
<td style="text-align: left;">Re-used PCR plate</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<p>281 samples were included in the analysis. On rack 154 the two samples from 2007-2008 were excluded, due to lack of DNA for the last treatment. Likewise for two of the positive controls on R156. In total, 21918 alleles were analysed, out of which 4645 (21.1926271%) were unscorable even after manual evaluation. We found a total of 446 mismatches on allelic level between all treatments across all racks. The genotype error rates varied from 0.002 to 0.047, with the error rates for the ‘Re-used PCR plate’ treatment being 10 times higher than the other treatments (see <a href="#tbl-allelemismatch">Table&nbsp;1</a>). A subset of the mismatches (266) were potential contamination cases (see <a href="#tbl-contamination">Table&nbsp;2</a>), predominately found on the ‘Re-used PCR plates’.</p>
<p><a href="#fig-Errorrates">Figure&nbsp;2</a></p>
<div id="fig-Errorrates" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/Errorrates.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;2: Mismatches and error rates</figcaption>
</figure>
</div>
<p>Two bayesian models with binomial distribution were fitted. The first model explored the treatment effect on match (0 mismatch, 1 match) per loci. The treatments were fit as indicator variables, excluding ‘standard procedure’, which was the reference treatment for the mismatching. ID, Rack, Loci and Mix were included as random effects.</p>
<p><strong>Match model</strong> <strong>Match ~ Internal control + Reused detection plate + Reused PCR plate + (1|ID) + (1|Rack) + (1|Loci) + (1|Mix)</strong></p>
<p>The second model explored the treatment effect on missing genotypes (0 missing, 1 present) per loci. The treatments were fit as indicator variables and ID, Rack, Loci and Mix were included as random effects.</p>
<p><strong>Missing genotypes model</strong> <strong>Missing genotypes ~ Standard procedure + Internal control + Reused detection plate + Reused PCR plate + (1|ID) + (1|Rack) + (1|Loci) + (1|Mix)</strong></p>
<p>The first model yielded a significant effect of all treatments, where the Reused detection plate treatment had a positively diverged posterior distribution and the Reused PCR plate treatment yielded a negatively diverged posterior distributed in relation to the Internal control treatment (See <a href="#fig-Models">Figure&nbsp;3</a> A).</p>
<p>The second model yielded a significant effect of all treatments, where the Internal control treatment and the Reused detection plate treatment had a positively diverged posterior distribution and the Reused PCR plate treatment yielded a negatively diverged posterior distributed in relation to the Standard procedure (See <a href="#fig-Models">Figure&nbsp;3</a> B).</p>
<div id="fig-Models" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/Densities.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;3: Posterior distribution from models</figcaption>
</figure>
</div>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
</section>
<section id="supporting-information" class="level1">
<h1>Supporting Information</h1>
<section id="description" class="level3">
<h3 class="anchored" data-anchor-id="description">Description</h3>
<p>Scientific research is a massive endeavor, consuming vast quantities of single-use plastics. The sustainable genetics project aim to empirically test the re-usability of lab plastic wear, specifically the plastic racks used for PCRs and detection. The project was based around the protocol for Antarctic fur seal genotyping using microsatellites. In this protocol, extracted DNA is first amplified using PCR and since transferred to detection plates for sequencing. The plates used for detection contain only a small diluted concentration of DNA, while the PCR plates have been used in the PCR process and therefore contain a high concentration of DNA. Therefore, we assume that the potential risk of contamination from a previously used plate would be higher in the PCR step.</p>
</section>
<section id="data" class="level3">
<h3 class="anchored" data-anchor-id="data">Data</h3>
<p>The data for this project are the raw sequencing reads from the ABI 3730xl capillary sequencer. Each files is identified with the following name structure: RackX_mixY_Z. Where X gives the rack number, Y the mix and Z the treatment (1 = standard procedure, 2 = internal control, 3 = Re-used detection plate and 4 = Re-used PCR plate).</p>
</section>
<section id="pcr-program" class="level3">
<h3 class="anchored" data-anchor-id="pcr-program">PCR Program</h3>
<div id="tbl-PCR" class="anchored">
<table class="table">
<caption>Table&nbsp;3: PCR program</caption>
<thead>
<tr class="header">
<th>Stage</th>
<th>No.&nbsp;of cycles</th>
<th>Temperature (°C)</th>
<th>Duration</th>
<th>Process</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>1</td>
<td>94</td>
<td>5 minutes</td>
<td>Heat up</td>
</tr>
<tr class="even">
<td>2</td>
<td>28</td>
<td>94</td>
<td>30 seconds</td>
<td>Denaturation</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>60/53</td>
<td>90 seconds</td>
<td>Annealing</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>72</td>
<td>30 seconds</td>
<td>Extension</td>
</tr>
<tr class="odd">
<td>3</td>
<td>1</td>
<td>60/53</td>
<td>30 minutes</td>
<td>Annealing</td>
</tr>
<tr class="even">
<td>4</td>
<td>1</td>
<td>10</td>
<td>Hold</td>
<td>Cool down</td>
</tr>
</tbody>
</table>
</div>
<p><em>Note: Annealing temperatures are mix-specific.</em></p>
</section>
<section id="missing-data" class="level3">
<h3 class="anchored" data-anchor-id="missing-data">Missing data</h3>
<div id="tbl-failed" class="cell tbl-parent quarto-layout-panel anchored">
<div class="panel-caption table-caption">
<p>Table&nbsp;4: Unscored alleles</p>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="tbl-failed-1" class="quarto-layout-cell quarto-layout-cell-subref anchored" data-ref-parent="tbl-failed" style="flex-basis: 100.0%;justify-content: center;">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<caption>(a) Rack 154</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Mix</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Standard procedure</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Internal control</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Reused detection plate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Reused PCR plate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Mix1</td>
<td style="text-align: right;">98</td>
<td style="text-align: right;">56</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">432</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mix2</td>
<td style="text-align: right;">58</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">154</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mix3</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">80</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mix4</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">544</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mix5</td>
<td style="text-align: right;">48</td>
<td style="text-align: right;">66</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">170</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total</td>
<td style="text-align: right;">278</td>
<td style="text-align: right;">194</td>
<td style="text-align: right;">128</td>
<td style="text-align: right;">1380</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">


</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="tbl-failed-2" class="quarto-layout-cell quarto-layout-cell-subref anchored" data-ref-parent="tbl-failed" style="flex-basis: 100.0%;justify-content: center;">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<caption>(b) Rack 155</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Mix</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Standard procedure</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Internal control</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Reused detection plate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Reused PCR plate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Mix1</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">50</td>
<td style="text-align: right;">66</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mix2</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">158</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mix3</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">112</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mix4</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">96</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mix5</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">132</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">430</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">251</td>
<td style="text-align: right;">110</td>
<td style="text-align: right;">862</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">


</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="tbl-failed-3" class="quarto-layout-cell quarto-layout-cell-subref anchored" data-ref-parent="tbl-failed" style="flex-basis: 100.0%;justify-content: center;">
<table data-quarto-postprocess="true" class="table table-sm table-striped small">
<caption>(c) Rack 156</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Mix</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Standard procedure</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Internal control</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Reused detection plate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Reused PCR plate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Mix1</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">66</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mix2</td>
<td style="text-align: right;">46</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">100</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mix3</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">84</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mix4</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">34</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mix5</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">48</td>
<td style="text-align: right;">400</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total</td>
<td style="text-align: right;">208</td>
<td style="text-align: right;">200</td>
<td style="text-align: right;">256</td>
<td style="text-align: right;">702</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">


</div>
</div>
</div>
<p><a href="#tbl-failed">Table&nbsp;4</a> display the number of unscored alleles within each mix on each plate. For R154 and R156 this is based on 93 individuals, while for R155 it is based on all 95 individuals on the plate.</p>
<p>Before starting the error rate evaluation, the no. of genotypes that could not be scored for each round and mix per rack is tabulated (See <a href="#tbl-failed">Table&nbsp;4</a>). From this, we know that a total of 4645 alleles could not be scored. In <a href="#fig-Heatmap">Figure&nbsp;4</a> and <a href="#fig-Heatmap2">Figure&nbsp;5</a>, the location of the missing data on the plate is visualized. More purple hues correspond to more missing loci for that particular samples, and we can conclude that certain samples seem to have failed almost completely. This could hint at poor DNA quality. The data is arranged in the plot to match the layout of the actual plate. Through visual inspection, there might be a slight tendency for the missing loci to be located more towards the edges of the plates compared to the center.</p>
<div id="fig-Heatmap" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/Heatmap.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;4: Heatmap of missing data before gapfilling</figcaption>
</figure>
</div>
<div id="fig-Heatmap2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/Heatmap2.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;5: Heatmap of missing data after gapfilling</figcaption>
</figure>
</div>
<p>Based on the heatmaps, it is obvious that even after gap filling, some individuals have a higher proportion of missing data than others. To further explore the distribution of missing data, the number of NAs for each row of the dataset was tabulated. Each row corresponds to one individual on one rack, which means that each individual is represented 4 times. Based on <a href="#fig-missingdata">Figure&nbsp;6</a> the majority of individuals have less than 10 errors, however 95 individuals are represented in the dataset with at least one occurrence of more than 10 missing alleles.</p>
<div id="fig-missingdata" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/Nasmissing.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;6: Missing data per individual</figcaption>
</figure>
</div>
<p>The genotyping error rate per allele and per locus was calculated across all treatments (<a href="#fig-Errorrates">Figure&nbsp;2</a>). To further understand where the errors occur in the dataset, the error rate per mix across all racks were calculated.</p>
<div id="fig-permixerrorrate" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/Errorratepermixplot.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;7: Error rate per mix</figcaption>
</figure>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>