<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ane Liv Berthelsen">
<meta name="dcterms.date" content="2023-11-20">

<title>Sustainable genetics project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="SGP_files/libs/clipboard/clipboard.min.js"></script>
<script src="SGP_files/libs/quarto-html/quarto.js"></script>
<script src="SGP_files/libs/quarto-html/popper.min.js"></script>
<script src="SGP_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="SGP_files/libs/quarto-html/anchor.min.js"></script>
<link href="SGP_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="SGP_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="SGP_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="SGP_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="SGP_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#description" id="toc-description" class="nav-link active" data-scroll-target="#description">Description</a>
  <ul class="collapse">
  <li><a href="#packages" id="toc-packages" class="nav-link" data-scroll-target="#packages">Packages</a></li>
  </ul></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data">Data</a></li>
  <li><a href="#analysis" id="toc-analysis" class="nav-link" data-scroll-target="#analysis">Analysis</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Sustainable genetics project</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Ane Liv Berthelsen </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 20, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="description" class="level1">
<h1>Description</h1>
<p>Scientific research is a massive endeavor, consuming vast quantities of single-use plastics. The sustainable genetics project aim to empirically test the re-usability of lab plastic wear, specifically the plastic racks used for PCRs and detection. The project was based around the protocol for Antarctic fur seal genotyping using microsatellites. In this protocol, extracted DNA is first amplified using PCR and since transfered to detection plates for sequencing. The plates used for detection contain only a small diluted concentration of DNA, while the PCR plates have been used in the PCR process and therefore contain a high concentration of DNA. Therefore, we assume that the potential risk of contamination from a previously used plate would be higher in the PCR step.</p>
<section id="packages" class="level3">
<h3 class="anchored" data-anchor-id="packages">Packages</h3>
</section>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<p>The 274 samples were collected from Bird Island, South Georgia (54°00024.800 S, 38°03004.100 W) during the austral summer of 2020-2021. Tissue samples were collected from the flipper or umbilical cord of Antarctic fur seal pups and stored in 20% dimethyl sulphoxide saturated with sodium chloride at -20°C. The DNA was extracted using a standard chloroform-isoamylalcohol protocol and genotyped at 39 microsatellite loci separated into 5 mixes for improved scoring. PCR amplification was performed using a Type It Kit (Qiagen) with the following program:</p>
<div id="tbl-PCR" class="anchored">
<table class="table">
<caption>Table&nbsp;1: PCR program</caption>
<thead>
<tr class="header">
<th>Stage</th>
<th>No.&nbsp;of cycles</th>
<th>Temperature (°C)</th>
<th>Duration</th>
<th>Process</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>1</td>
<td>94</td>
<td>5 minutes</td>
<td>Heat up</td>
</tr>
<tr class="even">
<td>2</td>
<td>28</td>
<td>94</td>
<td>30 seconds</td>
<td>Denaturation</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>60/53</td>
<td>90 seconds</td>
<td>Annealing</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>72</td>
<td>30 seconds</td>
<td>Extension</td>
</tr>
<tr class="odd">
<td>3</td>
<td>1</td>
<td>60/53</td>
<td>30 minutes</td>
<td>Annealing</td>
</tr>
<tr class="even">
<td>4</td>
<td>1</td>
<td>10</td>
<td>hold</td>
<td>Cool down</td>
</tr>
</tbody>
</table>
</div>
<p><em>Note: Annealing temperatures are mastermix-specific.</em></p>
<p>The PCR product was transferred to and diluted on detection plates before being resolved by electrophoresis on an ABI 3730xl capillary sequencer (Applied Biosystems, Waltham, MA, USA). Each plate contained three positive controls to ensure standardisation of microsatellite scoring across plates. Allele sizes were scored automatically using GeneMarker v. 2.6.2 (SoftGenetics, LLC., State College, PA, USA) and manually inspected and corrected when necessary.</p>
<p>To assess the re-usability of both PCR plates and detection plates, the samples were placed on a master plate from which the samples were transferred to PCR plates. The samples were distributed on a total of 3 master plates. The samples were subjected to four treatments: standard procedure, internal control, re-used PCR plate and re-used detection plate. This is illustrated by <a href="#fig-Schematic">Figure&nbsp;1</a>. The re-used plates all originated from the ‘standard procedure’ round. Thereby, we could ensure knowledge of the sample previously contained within a specific well. Following standard procedure, any locus with more than 20% missing data is gapfilled separately to minimize missing data.</p>
<div id="fig-Schematic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/SGP_setup.png" class="img-fluid figure-img" width="500"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Schematic overview</figcaption>
</figure>
</div>
<section id="washing-protocol" class="level4">
<h4 class="anchored" data-anchor-id="washing-protocol">Washing protocol</h4>
<p>After the initial use, the racks were washed using the following steps. First, each plate was individually rinsed with distilled water and emptied 10 times, before submerging in soap water for two hours. After soaking, each plate was again rinsed and emptied before left on a paper towel over night to dry.</p>
</section>
<section id="quantification-of-genotyping-errors" class="level4">
<h4 class="anchored" data-anchor-id="quantification-of-genotyping-errors">Quantification of genotyping errors</h4>
<p>To estimate the genotyping error rate, the ‘standard procedure’ treatment was first compared to the ‘internal control’ treatment to establish a baseline error rate. Any discrepancy between the two sets of genotypes was flagged as an error. The error rate was evaluated both for each allele and for the genotype on a locus-specific level and across all loci. After establishing a baseline error rate, the same comparision was preformed between the ‘standard procedure’ treatment and the two re-use treatments independently. In these cases, any discrepancy was first explored for the possibility of contamination, before flagged as an error, if no contamination was detectable.</p>
</section>
<section id="identification-of-contamination" class="level4">
<h4 class="anchored" data-anchor-id="identification-of-contamination">Identification of contamination</h4>
<p>The samples were each repeated for all four treatments. In the case of a discrepancy between the original genotype and the ‘re-used PCR plate’ or ‘re-used detection plate’ treatment, the new genotype was compared to the genotype of the individual previously genotyped in the same well. If the genotype matched the genotype of the original individual, the mis-matched was marked as contamination.</p>
</section>
<section id="data" class="level3">
<h3 class="anchored" data-anchor-id="data">Data</h3>
<p>The data for this project are the raw sequencing reads from the ABI 3730xl capillary sequencer. Each files is identified with the following name structure: RackX_mixY_Z. Where X gives the rack number, Y the mix and Z the treatment (1 = standard procedure, 2 = internal control, 3 = Re-used PCR plate and 4 = Re-used detection plate).</p>
<p>Before starting the error rate evaluation, the no. of genotypes that could no be scored for each round and mix per rack is tabulated (See <a href="#tbl-failed">Table&nbsp;2</a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-failed" class="anchored">
<table class="table table-sm table-striped small">
<caption>Table&nbsp;2: Failed genotypes</caption>
<colgroup>
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Mix</th>
<th style="text-align: right;">R154_R1</th>
<th style="text-align: right;">R154_R2</th>
<th style="text-align: right;">R154_R3</th>
<th style="text-align: right;">R154_R4</th>
<th style="text-align: right;">R155_R1</th>
<th style="text-align: right;">R155_R2</th>
<th style="text-align: right;">R155_R3</th>
<th style="text-align: right;">R155_R4</th>
<th style="text-align: right;">R156_R1</th>
<th style="text-align: right;">R156_R2</th>
<th style="text-align: right;">R156_R3</th>
<th style="text-align: right;">R156_R4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Mix1</td>
<td style="text-align: right;">108</td>
<td style="text-align: right;">88</td>
<td style="text-align: right;">58</td>
<td style="text-align: right;">1014</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">50</td>
<td style="text-align: right;">66</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">66</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mix2</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">162</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">158</td>
<td style="text-align: right;">46</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">60</td>
<td style="text-align: right;">102</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mix3</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">58</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">108</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">112</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">50</td>
<td style="text-align: right;">84</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mix4</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">32</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">548</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">96</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">1140</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mix5</td>
<td style="text-align: right;">64</td>
<td style="text-align: right;">78</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">186</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">224</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">430</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">70</td>
<td style="text-align: right;">406</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total</td>
<td style="text-align: right;">308</td>
<td style="text-align: right;">296</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">2018</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">343</td>
<td style="text-align: right;">114</td>
<td style="text-align: right;">862</td>
<td style="text-align: right;">208</td>
<td style="text-align: right;">1306</td>
<td style="text-align: right;">276</td>
<td style="text-align: right;">710</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>In <a href="#fig-Heatmap">Figure&nbsp;2</a>, the location of the missing data on the plate is visualized. More purple hues correspond to more missing loci for that particular samples, and we can conclude that certain samples seem to have failed almost completely. This could hint at poor DNA quality. The data is arranged in the plot to match the layout of the actual plate. Through visual inspection, there might be a slight tendency for the missing loci to be located more towards the edges of the plates compared to the center.</p>
<div id="fig-Heatmap" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Results/Figures/Heatmap.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;2: Heatmap of missing data</figcaption>
</figure>
</div>
<section id="error-rate-calculation" class="level4">
<h4 class="anchored" data-anchor-id="error-rate-calculation">Error rate calculation</h4>
<p>The error rate is calculated by comparing all scored alleles and genotypes from the standard procedure within one rack with the other treatments starting with the internal control. The internal control is included as a baseline for error rates to have a comparable metric to evaluate if the reused plates inflate the error rate.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-allelemismatch" class="anchored">
<table class="table table-sm table-striped small">
<caption>Table&nbsp;3: Allelic mismatches</caption>
<colgroup>
<col style="width: 10%">
<col style="width: 10%">
<col style="width: 31%">
<col style="width: 23%">
<col style="width: 24%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Rack 1</th>
<th style="text-align: left;">Rack 2</th>
<th style="text-align: left;">Treatment</th>
<th style="text-align: right;">No.&nbsp;of mismatches</th>
<th style="text-align: right;">Allelic error rate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">R154_R1</td>
<td style="text-align: left;">R154_R2</td>
<td style="text-align: left;">Internal control</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">0.0012890</td>
</tr>
<tr class="even">
<td style="text-align: left;">R155_R1</td>
<td style="text-align: left;">R155_R2</td>
<td style="text-align: left;">Internal control</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">0.0029825</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R156_R1</td>
<td style="text-align: left;">R156_R2</td>
<td style="text-align: left;">Internal control</td>
<td style="text-align: right;">37</td>
<td style="text-align: right;">0.0060875</td>
</tr>
<tr class="even">
<td style="text-align: left;">R154_R1</td>
<td style="text-align: left;">R154_R3</td>
<td style="text-align: left;">Re-used PCR</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">0.0024453</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R155_R1</td>
<td style="text-align: left;">R155_R3</td>
<td style="text-align: left;">Re-used PCR</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0.0013759</td>
</tr>
<tr class="even">
<td style="text-align: left;">R156_R1</td>
<td style="text-align: left;">R156_R3</td>
<td style="text-align: left;">Re-used PCR</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">0.0043613</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R154_R1</td>
<td style="text-align: left;">R154_R4</td>
<td style="text-align: left;">Re-used detection plate</td>
<td style="text-align: right;">133</td>
<td style="text-align: right;">0.0250094</td>
</tr>
<tr class="even">
<td style="text-align: left;">R155_R1</td>
<td style="text-align: left;">R155_R4</td>
<td style="text-align: left;">Re-used detection plate</td>
<td style="text-align: right;">114</td>
<td style="text-align: right;">0.0174793</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R156_R1</td>
<td style="text-align: left;">R156_R4</td>
<td style="text-align: left;">Re-used detection plate</td>
<td style="text-align: right;">74</td>
<td style="text-align: right;">0.0111952</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="inspect-mismatches-for-contamination" class="level4">
<h4 class="anchored" data-anchor-id="inspect-mismatches-for-contamination">Inspect mismatches for contamination</h4>
<p>The comparison between the different treatments and the standard procedure yielded a list of mismatches. To identify whether these mismatches in the case of the ‘re-used PCR plate’ and ‘re-used detection plate’ treatment was due to contamination from the samples previous found in the same well, the new genotype was compared to the original genotype scored from that well. If the genotype matched the original genotype, the mis-match was marked as potential case of contamination.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-contamination" class="anchored">
<table class="table table-sm table-striped small">
<caption>Table&nbsp;4: Potential contamination cases</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Rack</th>
<th style="text-align: left;">Control</th>
<th style="text-align: right;">Cases</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">R154_R3</td>
<td style="text-align: left;">R156_R1</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">R154_R4</td>
<td style="text-align: left;">R156_R1</td>
<td style="text-align: right;">98</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R155_R3</td>
<td style="text-align: left;">R154_R1</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">R155_R4</td>
<td style="text-align: left;">R154_R1</td>
<td style="text-align: right;">96</td>
</tr>
<tr class="odd">
<td style="text-align: left;">R156_R3</td>
<td style="text-align: left;">R155_R1</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="even">
<td style="text-align: left;">R156_R4</td>
<td style="text-align: left;">R155_R1</td>
<td style="text-align: right;">54</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
</section>
<section id="analysis" class="level3">
<h3 class="anchored" data-anchor-id="analysis">Analysis</h3>
<p>To determine if treatment had a significant effect on the following three measures; no. of mismatches, allelic error rate and potential cases of contamination, we performed ANOVAs followed by a post hoc test for pairwise comparisons for the three treatment groups measured against the standard procedure treatment.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>            Df Sum Sq Mean Sq F value Pr(&gt;F)   
Treatment    2  14863    7431   18.29 0.0028 **
Residuals    6   2437     406                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = `No. of mismatches` ~ Treatment, data = Mismatches)

$Treatment
                                              diff        lwr       upr
Re-used detection plate-Internal control  84.66667   34.17376 135.15958
Re-used PCR-Internal control              -3.00000  -53.49291  47.49291
Re-used PCR-Re-used detection plate      -87.66667 -138.15958 -37.17376
                                             p adj
Re-used detection plate-Internal control 0.0051045
Re-used PCR-Internal control             0.9819009
Re-used PCR-Re-used detection plate      0.0042908</code></pre>
</div>
</div>
<p>The total no. of mismatches between the standard procedure and the additional treatments could be significantly determined by treatment. Further exploration using a Tukey pairwise comparison revealed that this treatment effect was driven by a significant difference in no. of mismatches in the ‘re-used detection plate’ treatment when compared to both the internal control and the re-used PCR treatment. When comparing those two treatments to eachother, no significant difference could be found. The same was true for the error rate.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>            Df    Sum Sq   Mean Sq F value  Pr(&gt;F)   
Treatment    2 0.0004391 2.196e-04   11.75 0.00841 **
Residuals    6 0.0001121 1.868e-05                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = `Allelic error rate` ~ Treatment, data = Mismatches)

$Treatment
                                                  diff          lwr
Re-used detection plate-Internal control  0.0144415910  0.003613101
Re-used PCR-Internal control             -0.0007255239 -0.011554014
Re-used PCR-Re-used detection plate      -0.0151671149 -0.025995605
                                                  upr     p adj
Re-used detection plate-Internal control  0.025270081 0.0151155
Re-used PCR-Internal control              0.010102966 0.9770594
Re-used PCR-Re-used detection plate      -0.004338625 0.0120883</code></pre>
</div>
</div>
<p>To explore the effect of treatment on the number of potential contamination cases, a non-parametric wilcoxon test was used.</p>
<p><em>Note: the test is performed on only one case of the ‘re-used detection plate’ treatment and therefore not yet considered informative.</em></p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon rank sum exact test

data:  Cases by Treatment
W = 6, p-value = 0.5333
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
</div>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<p>Will come later</p>
<p><a href="#fig-allelicerrors">Figure&nbsp;3</a> shows the total no. of mismatches observed when comparing the standard procedure treatment of each rack with the other treatments.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-allelicerrors" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="SGP_files/figure-html/fig-allelicerrors-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;3: Allelic mismatches</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-errorrate" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="SGP_files/figure-html/fig-errorrate-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;4: Allelic error rate</figcaption>
</figure>
</div>
</div>
</div>
<p>We found a total of 446 mismatches between all treatments. A subset of these (262) were potential contamination cases.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-contamination" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="SGP_files/figure-html/fig-contamination-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;5: Potential contamination</figcaption>
</figure>
</div>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>